query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -353,53 +349,46 @@ object LogicalPlan {
    }
   }
 
-  def getLabelValueFromLogicalPlan(logicalPlan: LogicalPlan, labelName: String): Option[Seq[String]] = {
-    getAllLabelValueFromGroups(getLabelValueOperatorsFromLogicalPlan(logicalPlan), labelName)
+  def getColumnValues(logicalPlan: LogicalPlan, labelName: String): Set[String] = {
+    getColumnValues(getColumnFilterGroup(logicalPlan), labelName)
   }
 
-  def getAllLabelValueFromGroups(labelValues: Option[Seq[LabelValueOperatorGroup]],
-                                   labelName: String): Option[Seq[String]] = {
-    labelValues match {
-      case None => None
-      case _    => labelValues.get.flatMap(group => getLabelValueFromGroup(labelName, group)).distinct match {
-        case Nil                    => None
-        case lVFilters: Seq[String] => Some(lVFilters)
-      }
+  def getColumnValues(columnFilterGroup: Seq[Set[ColumnFilter]],
+                      labelName: String): Set[String] = {
+    columnFilterGroup.flatMap (columnFilters => getColumnValues(columnFilters, labelName)) match {
+      case columnValues: Iterable[String] => if (columnValues.isEmpty) Set.empty else columnValues.toSet
+      case _ => Set.empty
     }
   }
 
-  def getLabelValueFromGroup(labelName: String, group: LabelValueOperatorGroup): Seq[String] = {
-    group.labelValueOperators.flatMap(lops => {
-      lops.columnName.equals(labelName) match {
-        case true => lops.value
-        case false => Seq()
+  def getColumnValues(columnFilters: Set[ColumnFilter], labelName: String): Set[String] = {
+    columnFilters.flatMap(cFilter => {
+      cFilter.column == labelName match {
+        case true => cFilter.filter.valuesStrings.map(_.toString)
+        case false => Seq.empty
```

Comment:
Set.empty?

Reply:
Inside the map it is returning Seq. And flatmap will convert it to Set